tests/test_jobqueue.py

```diff
@@ -291,20 +291,46 @@ def test_run_monthly(self, job_queue):
 
         day = date_time.day
         next_job_day = date_time + dtm.timedelta(calendar.monthrange(
-                                                 date_time.year, date_time.month)[1])
+            date_time.year, date_time.month)[1])
         expected_reschedule_time += calendar.monthrange(date_time.year,
                                                         date_time.month)[1] * 24 * 60 * 60
-        while next_job_day.day != day:
-            next_job_day += dtm.timedelta(calendar.monthrange(
-                                          next_job_day.year, next_job_day.month))
-            expected_reschedule_time += calendar.monthrange(date_time.year,
-                                                            date_time.month)[1] * 24 * 60 * 60
+        if next_job_day.day != day:
+            # account for when next month does not have the specified date
+            day_difference = day - next_job_day.day
+            next_job_day += dtm.timedelta(day_difference)
+            expected_reschedule_time += day_difference * 24 * 60 * 60
 
         job_queue.run_monthly(self.job_run_once, time_of_day, day)
         sleep(0.2)
         assert self.result == 1
         assert job_queue._queue.get(False)[0] == pytest.approx(expected_reschedule_time)
 
+    def test_run_monthly_and_not_strict(self, job_queue):
+        delta, now = 0.1, time.time()
+        date_time = dtm.datetime.utcfromtimestamp(now)
+        time_of_day = (date_time + dtm.timedelta(seconds=delta)).time()
+        expected_reschedule_time = now + delta
+
+        day = date_time.day
+        date_time += dtm.timedelta(calendar.monthrange(date_time.year,
+                                                       date_time.month)[1] - day)
+        # next job should be scheduled on last day of month if day_is_strict is True
+        expected_reschedule_time += (calendar.monthrange(date_time.year,
+                                                         date_time.month)[1] - day) * 24 * 60 * 60
+
+        if day == calendar.monthrange(date_time.year, date_time.month)[1]:
+            # next job should be scheduled at the end of the next month if this test is run on
```

Comment:
No, only if the test is run *just* before mitnight on the months last day, so that `time_of_day` is like 00:00:00.05 on the next months first day. I think we can neglect that case an hence remove this if clause. Please test this behaviour manually by replacing the `time.time()` above with something like `datetime.datetime(2020,1,31).timestamp()`.

Reply:
done!